app/models/package.rb

```diff
@@ -18,29 +18,96 @@ def to_param
   validates :bag_id, presence: true, length: { minimum: 6 }
   validates :user_id, presence: true
   validates :external_id, presence: true
+  validates :format, presence: true
+
+  class Format < String
+    class Bag < Format
+      def initialize
+        super("bag")
+      end
+    end
+
+    def self.bag
```

Comment:
perhaps this could just be a symbol

Reply:
or an enum or something - but either way it is not bound to the storage proxy class directly

Reply:
I'm using Chipmunk::Bag.format now. If we need a registry of known storage formats, volumes, etc., we can do that separately.